### PR TITLE
feat(TG-277): timeseries

### PR DESCRIPTION
**Pull request by vreshch**
mock backend for timeseries

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-backend/pull/71" target="_blank">ReviewNinja

File: src/API/timeseries/timeseries.controller.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        const err = new APIError('Period is not provided to wrong', httpStatus.BAD_REQUEST);
+        return next(err);
+    }
+
+    if (!req.query.sensor) {
+        const err = new APIError('Sensor is not provided', httpStatus.BAD_REQUEST);
+        return next(err);
+    }
+
+    const sensor = req.query.sensor;
```
Comment (velmyk) [inline src/API/timeseries/timeseries.controller.js]: Please move vars declarations to the top of function body

File: src/API/timeseries/timeseries.controller.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
+const PERIOD_TO_INTERVAL = {
+    'day': 60,
+    'week': 60,
+    'month': 60*24,
+    'year': 60*24
+};
+
+import APIError from '../helpers/APIError';
```
Comment (velmyk) [inline src/API/timeseries/timeseries.controller.js]: It is better to place import statements on top of the file

File: src/API/timeseries/timeseries.test.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import request from 'supertest-as-promised';
```
Comment (velmyk) [inline src/API/timeseries/timeseries.test.js]: What about unit tests? MB better to move logic to service?

File: src/API/timeseries/timeseries.controller.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        return next(err);
+    }
+
+    const sensor = req.query.sensor;
+    const period = req.query.period;
+    const stepMin = PERIOD_TO_INTERVAL[period];
+    const from = moment();
+    const to = from.clone();
+
+    // Period
```
Comment (ashkanhosseini) [inline src/API/timeseries/timeseries.controller.js]: I'd move this to a private function with a meaningful name

File: src/API/timeseries/timeseries.controller.js (hunk 1 above)
Comment (vreshch) [inline src/API/timeseries/timeseries.controller.js]: the const and let declarations is block scoped, therefor decided to keep it like it is

File: src/API/timeseries/timeseries.test.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    it('should be available and give error by default', (done) => {
+        request(app)
+        .get("/api/timeseries")
+        .expect(httpStatus.BAD_REQUEST)
+        .then(() => {
+            done()
+         });
+    });
+
+    it('should give error by default', (done) => {
```
Comment (vreshch) [inline src/API/timeseries/timeseries.test.js]: thx. fixed

File: src/API/timeseries/timeseries.controller.js (hunk 2 above)
Comment (vreshch) [inline src/API/timeseries/timeseries.controller.js]: done

File: src/API/timeseries/timeseries.test.js (hunk 3 above)
Comment (vreshch) [inline src/API/timeseries/timeseries.test.js]: make sense, but it's just mock data for now and will be rewritten

File: src/API/timeseries/timeseries.controller.js (hunk 1 above)
Comment (vreshch) [inline src/API/timeseries/timeseries.controller.js]: done

File: src/API/timeseries/timeseries.controller.js (hunk 4 above)
Comment (vreshch) [inline src/API/timeseries/timeseries.controller.js]: agree, but for now it's just mock data

**Pull request merged**